support-frontend/assets/pages/digital-subscription-checkout/components/datePicker/datePicker.jsx

```diff
@@ -0,0 +1,256 @@
+    const dateIsNotADate = !DateUtils.isDate(date);
+    const dateOutsideRange = this.dateIsOutsideRange(date);
+    const dateIsPast = this.dateIsPast(date);
+    const eightyNineDaysFromNow = this.getLatestAvailableDateText();
+    this.setState({ dateValidated: true, dateError: '' });
+    if (dateIsNotADate) {
+      this.setState({
+        dateError: 'No date has been selected as the date is not valid. Please try again',
+        day: '',
+        month: '',
+        year: '',
+      });
+      this.updateStartDate();
+    } else if (dateOutsideRange) {
+      this.setState({
+        dateError: `No date has been recorded as the date entered was not available. Please enter a date up to ${eightyNineDaysFromNow}`,
+        day: '',
+        month: '',
+        year: '',
+      });
+      this.updateStartDate();
+    } else if (dateIsPast) {
+      this.setState({
+        dateError: `No date has been recorded as the date was in the past. Please enter a date between today and ${eightyNineDaysFromNow}`,
+        day: '',
+        month: '',
+        year: '',
+      });
+      this.updateStartDate();
+    }
+  }
+
+  handleCalendarDate = (date: Date) => {
+    if (this.dateIsPast(date) || this.dateIsOutsideRange(date)) {
+      return;
+    }
+    const dateArray = formatMachineDate(date).split('-');
+    this.setState({
+      dateError: '',
+      dateValidated: false,
+      day: dateArray[2],
+      month: dateArray[1],
+      year: dateArray[0],
+    }, this.updateStartDate);
+  }
+
+  handleInput = (value: string, field: string) => {
+    if (/^[0-9]+$/.test(value) === true) {
+      this.setState({ [field]: value, dateError: '', dateValidated: false }, this.updateStartDate);
+    } else {
+      this.setState({ [field]: '', dateError: '', dateValidated: false }, this.updateStartDate);
+    }
+  }
+
+  updateStartDate = () => this.props.onChange(`${this.state.day}/${this.state.month}/${this.state.year}`)
+
+  render() {
+    const { state } = this;
+    const { value } = this.props;
+    const today = Date.now();
+    const currentMonth = new Date(today);
+    const threeMonthRange = DateUtils.addMonths(currentMonth, 3);
+    const valueArray = value ? value.split('/') : [];
+    const valueDate = valueArray.length ? new Date(`${valueArray[1]} ${valueArray[0]} ${valueArray[2]}`) : null;
+
+    return (
+      <div>
+        <fieldset css={startDateGroup} role="group" aria-describedby="date-hint">
+          <p id="date-hint">
+            {`Please choose a date up to ${this.getLatestAvailableDateText()} for your gift to be emailed to the recipient.`}
+          </p>
+          <div css={startDateFields}>
+            <div css={inputLayoutWithMargin}>
+              <InputWithLabel
+                label="Day"
+                value={state.day}
+                onChange={e => this.handleInput(e.target.value, 'day')}
+                minLength={1}
+                maxLength={2}
+                type="text"
```

Comment:
Currently the Day/Month/Year inputs don't seem to be correctly associated with their labels- I think we need to pass an 'id' prop here.

Reply:
Hmm, sorry - I don't really follow what you mean here. Could you explain a little more?

Reply:
Sorry, I wasn't super clear! The inputs for day/month/year currently don't have `id` attributes that correspond to `for` attributes on their respective labels, which makes the screen reader experience less clear- when you're on the input field, the date-hint message is read but not the label for the specific field that you're on, and if you open the form controls rotor with ctrl+opt+u the labels for the fields aren't shown there.

I think to fix this you just need to pass an `id` prop to the `InputWithLabel` component.

Reply:
Thanks Imogen - I thinks this is fixed now